misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/TesterinaFunction.java

```diff
@@ -118,18 +117,12 @@ private static Object runOnSchedule(Class<?> initClazz, String name, Scheduler s
             scheduler.start();
             final Throwable t = out.getPanic();
             final Object result = out.getResult();
-            if (result instanceof BError) {
-                throw new BallerinaTestException((BError) result);
-            }
-            if (result instanceof Exception) {
-                throw new BallerinaTestException((Exception) result);
-            }
             if (t != null) {
                 throw new BallerinaTestException("Error while invoking function '" + funcName + "'", t.getMessage());
```

Comment:
We'll have to return this exception too right? To keep it consistent.

Reply:
Changed it to return the exception to be handled by the BTestRunner